src/diagnostics/MOM_spatial_means.F90

```diff
@@ -279,44 +302,54 @@ function global_volume_mean(var, h, G, GV, scale, tmp_scale, unscale)
-                                               !! variable that is reversed in the return value [a A-1 ~> 1]
+                                               !! variable that is reversed in the return value [a A-1 ~> 1],
+                                               !! or [b B-1 ~> 1] if unscale is also present.
   real,          optional, intent(in)  :: unscale !< A rescaling factor for the variable [a A-1 ~> 1]
                                                !! that converts it back to unscaled (e.g., mks)
-                                               !! units to enable the use of the reproducing sums.
+                                               !! units to enable the use of the reproducing sums, or
+                                               !! a factor converting between rescaled units if
+                                               !! tmp_scale is also present [B A-1 ~> b a-1].
                                                !! Here scale and unscale are synonymous, but unscale
                                                !! is preferred and takes precedence if both are present.
-  real :: global_volume_mean  !< The thickness-weighted average of var in the arbitrary scaled [A] or
+  real :: global_volume_mean  !< The thickness-weighted average of var in the arbitrary scaled [A ~> a] or [B ~> b] or
                               !! unscaled [a] units of var, depending on which optional arguments are provided
 
   ! Local variables
-  ! In the following comments, [A] is used to indicate the arbitrary, possibly rescaled units of the
+  ! In the following comments, [A ~> a] is used to indicate the arbitrary, possibly rescaled units of the
   ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums
-  real :: temp_scale ! A temporary scaling factor [a A-1 ~> 1] or [1]
-  real :: scalefac   ! A scaling factor for the variable [a A-1 ~> 1]
-  real :: weight_here ! The volume or mass of a grid cell [m3] or [kg]
-  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! The volume integral of the variable in a column [a m3] or [a kg]
-  real, dimension(SZI_(G),SZJ_(G)) :: sum_weight  ! The volume or mass of each column of water [m3] or [kg]
+  ! [A ~> a] and [B ~> b] are the same units unless tmp_scale and unscale are both present.
+  real :: temp_scale ! A temporary scaling factor [a A-1 ~> 1] or [b B-1 ~> 1] or [1]
+  real :: scalefac   ! A scaling factor for the variable [a A-1 ~> 1] or [B A-1 ~> b a-1] or [1]
+  real :: weight_here ! The volume or mass of a grid cell [L2 m ~> m3] or [L2 kg m-2 ~> kg]
+  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! The volume or mass integral of the variable in a column
+                                                  ! [B L2 m ~> b m3] or [B L2 kg m-2 ~> b kg] or
+                                                  ! [L2 a m ~> a m3] or [L2 a kg m-2 ~> a kg]
```

Comment:
Same as above, there should be six units for `tmpForSumming` (missing the two when `scalefac` is [1]).

Reply:
I have added a table in a comment document all of these various units and when they occur.